uptime_service_validation/coordinator/helper.py

```diff
@@ -1,84 +1,281 @@
+        finally:
+            cursor.close()
+        self.logger.info("create_point_record  end ")
+        return 0
+
+    def update_scoreboard(self, score_till_time, uptime_days=30):
+        "Update the block producer scores."
+        self.logger.info("updateScoreboard  start ")
+        sql = """with vars  (snapshot_date, start_date) as( values (%s AT TIME ZONE 'UTC',
+                        (%s - interval '%s' day) AT TIME ZONE 'UTC')
+              )
+              , epochs as(
+                select extract('epoch' from snapshot_date) as end_epoch,
+                extract('epoch' from start_date) as start_epoch from vars
+              )
+              , b_logs as(
+                select (count(1) ) as surveys
+                from bot_logs b , epochs e
+                where b.batch_start_epoch >= start_epoch and  b.batch_end_epoch <= end_epoch
+              )
+              , scores as (
+                select p.node_id, count(p.bot_log_id) bp_points
+                from points p join bot_logs b on p.bot_log_id =b.id, epochs
+                where b.batch_start_epoch >= start_epoch and  b.batch_end_epoch <= end_epoch
+                group by 1
+              )
+              , final_scores as (
+                select node_id, bp_points,
+                surveys, trunc( ((bp_points::decimal*100) / surveys),2) as score_perc
+                from scores l join nodes n on l.node_id=n.id, b_logs t
+              )
+              update nodes nrt set score = s.bp_points, score_percent=s.score_perc
+              from final_scores s where nrt.id=s.node_id """
+
+        history_sql = """insert into score_history (node_id, score_at, score, score_percent)
+                      SELECT id as node_id, %s, score, score_percent from nodes where score is not null """
+        try:
+            cursor = self.connection.cursor()
+            cursor.execute(
+                sql,
+                (
+                    score_till_time,
+                    score_till_time,
+                    uptime_days,
+                ),
+            )
+            cursor.execute(history_sql, (score_till_time,))
+        except (Exception, psycopg2.DatabaseError) as error:
+            self.logger.error(ERROR.format(error))
+            cursor.close()
+            return -1
+        finally:
+            cursor.close()
+        self.logger.info("updateScoreboard  end ")
+        return 0
+
+    def get_existing_nodes(self):
+        "Get the list of all known block producers."
+        cursor = self.connection.cursor()
+        try:
+            cursor.execute("select block_producer_key from nodes")
+            result = cursor.fetchall()
+            nodes = pd.DataFrame(result, columns=["block_producer_key"])
+        except (Exception, psycopg2.DatabaseError) as error:
+            self.logger.error(ERROR.format(error))
+            return -1
+        finally:
+            cursor.close()
+        return nodes
+
+
+def getRelations(df):
```

Comment:
since other method names (rightfully) renamed to snake case, then let's perhaps rename all.

Reply:
Done.